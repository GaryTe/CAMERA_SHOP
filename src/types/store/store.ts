import { store } from '../../store/store/store';
import { listCamera, CardProduct, Reviews, PromoProduct, Review } from '../types-response/types-response';

export type Cameras = {
  cameras: listCamera | null;
  modifiedListProduct: listCamera | null;
}

export type CardCamera = {
  dataCamera: CardProduct | null;
}

export type SimilaCameras = {
  dataSimilaCameras: listCamera | null;
  sortListCamerra: listCamera | null;
  counterMin: number;
  counterMax: number;
}

export type ReviewsCamera = {
  dataReviews: Reviews | Array<Review>;
  resultSortReviews: Reviews | null;
  reviews: Reviews | null;
  counter: number;
}

export type PromoCamera = {
  dataPromoCamera: PromoProduct | null;
}

export type Cart = {
  dataProduct: listCamera | Array<CardProduct>;
  loading: boolean;
}

export type RootState = ReturnType<typeof store.getState>;
