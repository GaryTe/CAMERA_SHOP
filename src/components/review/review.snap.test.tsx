import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import {useState} from 'react';
import Review from './review';
import { store } from '../../store/store/store';

const mockUseState = jest.fn(() => useState(false));

const history = createMemoryHistory();

const children = (
  <Review
    cameraId={'1'}
    onSetModalWindow={mockUseState}
    onSetModalWindowMessage={mockUseState}
  />
);

describe('Component: Review', () => {
  it('should render Review', () => {

    const {container} = render(
      <Provider store={store}>
        <Router
          location={history.location}
          navigationType={history.action}
          navigator={history}
        >
          {children}
        </Router>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
