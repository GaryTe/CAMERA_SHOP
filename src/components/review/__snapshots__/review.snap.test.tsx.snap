// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MessageAboutAddProduct should render MessageAboutAddProduct 1`] = `
<div>
  <div
    class="modal is-active"
  >
    <div
      class="modal__wrapper"
    >
      <div
        class="modal__overlay"
      />
      <div
        class="modal__content"
      >
        <p
          class="title title--h4"
        >
          Оставить отзыв
        </p>
        <div
          class="form-review"
        >
          <form
            method="post"
          >
            <div
              class="form-review__rate"
            >
              <fieldset
                class="rate form-review__item"
              >
                <legend
                  class="rate__caption"
                >
                  Рейтинг
                  <svg
                    aria-hidden="true"
                    height="9"
                    width="9"
                  >
                    <use
                      xlink:href="#icon-snowflake"
                    />
                  </svg>
                </legend>
                <div
                  class="rate__bar"
                >
                  <div
                    class="rate__group"
                    name="rate"
                  >
                    <input
                      class="visually-hidden"
                      id="star-5"
                      name="rate"
                      type="radio"
                      value="5"
                    />
                    <label
                      class="rate__label"
                      for="star-5"
                      title="Отлично"
                    />
                    <input
                      class="visually-hidden"
                      id="star-4"
                      name="rate"
                      type="radio"
                      value="4"
                    />
                    <label
                      class="rate__label"
                      for="star-4"
                      title="Хорошо"
                    />
                    <input
                      class="visually-hidden"
                      id="star-3"
                      name="rate"
                      type="radio"
                      value="3"
                    />
                    <label
                      class="rate__label"
                      for="star-3"
                      title="Нормально"
                    />
                    <input
                      class="visually-hidden"
                      id="star-2"
                      name="rate"
                      type="radio"
                      value="2"
                    />
                    <label
                      class="rate__label"
                      for="star-2"
                      title="Плохо"
                    />
                    <input
                      class="visually-hidden"
                      id="star-1"
                      name="rate"
                      type="radio"
                      value="1"
                    />
                    <label
                      class="rate__label"
                      for="star-1"
                      title="Ужасно"
                    />
                  </div>
                  <div
                    class="rate__progress"
                  >
                    <span
                      class="rate__stars"
                    >
                      0
                    </span>
                    <span>
                      /
                    </span>
                    <span
                      class="rate__all-stars"
                    >
                      5
                    </span>
                  </div>
                </div>
                <p
                  class="rate__message"
                />
              </fieldset>
              <div
                class="custom-input form-review__item"
              >
                <label>
                  <span
                    class="custom-input__label"
                  >
                    Ваше имя
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <input
                    name="user-name"
                    placeholder="Введите ваше имя"
                    type="text"
                  />
                </label>
                <p
                  class="custom-input__error"
                />
              </div>
              <div
                class="custom-input form-review__item"
              >
                <label>
                  <span
                    class="custom-input__label"
                  >
                    Достоинства
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <input
                    name="user-plus"
                    placeholder="Основные преимущества товара"
                    type="text"
                  />
                </label>
                <p
                  class="custom-input__error"
                />
              </div>
              <div
                class="custom-input form-review__item"
              >
                <label>
                  <span
                    class="custom-input__label"
                  >
                    Недостатки
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <input
                    name="user-minus"
                    placeholder="Главные недостатки товара"
                    type="text"
                  />
                </label>
                <p
                  class="custom-input__error"
                />
              </div>
              <div
                class="custom-textarea form-review__item"
              >
                <label>
                  <span
                    class="custom-textarea__label"
                  >
                    Комментарий
                    <svg
                      aria-hidden="true"
                      height="9"
                      width="9"
                    >
                      <use
                        xlink:href="#icon-snowflake"
                      />
                    </svg>
                  </span>
                  <textarea
                    minlength="5"
                    name="user-comment"
                    placeholder="Поделитесь своим опытом покупки"
                  />
                </label>
                <div
                  class="custom-textarea__error"
                />
              </div>
            </div>
            <button
              class="btn btn--purple form-review__btn"
              type="submit"
            >
              Отправить отзыв
            </button>
          </form>
        </div>
        <button
          aria-label="Закрыть попап"
          class="cross-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            height="10"
            width="10"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
